packages/jira-adapter/src/filters/automation/automation_structure.ts

```diff
@@ -56,28 +59,132 @@ const removeInnerIds = async (instance: InstanceElement): Promise<void> => {
   })).value
 }
 
+const replaceStringValuesFieldName = async (instance: InstanceElement): Promise<void> => {
+  instance.value = (await transformElement({
+    element: instance,
+    strict: false,
+    allowEmpty: true,
+    transformFunc: async ({ value, field }) => {
+      if (_.isPlainObject(value)
+      && (await field?.getType())?.elemID.typeName === AUTOMATION_COMPONENT_TYPE
+      && _.isString(value.value)) {
+        value.rawValue = value.value
+        value.value = undefined
+      }
+      return value
+    },
+  })).value
+}
+
+// linkType field is a string containing a reference to IssueLinkType and the link direction
+// we separate the field in order to resolve the reference
+const separateLinkTypeField = async (instance: InstanceElement): Promise<void> => {
+  instance.value = (await transformElement({
+    element: instance,
+    strict: false,
+    allowEmpty: true,
+    transformFunc: async ({ value, path }) => {
+      if (_.isPlainObject(value)
+      && path?.name === 'value' && _.isString(value.linkType)) {
+        const [linkTypeDirection, linkTypeId] = _.split(value.linkType, ':')
+        if (linkTypeDirection !== undefined && linkTypeId !== undefined) {
+          value.linkType = linkTypeId
+          value.linkTypeDirection = linkTypeDirection
+        }
+      }
+      return value
+    },
+  })).value
+}
+
+const consolidateLinkTypeFields = async (instance: InstanceElement): Promise<void> => {
+  instance.value = (await transformElement({
+    element: instance,
+    strict: false,
+    allowEmpty: true,
+    transformFunc: async ({ value, path }) => {
+      if (_.isPlainObject(value) && path?.name === 'value'
+      && value.linkType !== undefined && value.linkTypeDirection !== undefined) {
```

Comment:
nits:
1. maybe extract to a typeguard?
2. indentation nit - this should not be directly below the `if` as it is part of the condition. one option (though it might become irrelevant after 1):
```suggestion
      if (
        _.isPlainObject(value)
        && path?.name === 'value'
        && value.linkType !== undefined
        && value.linkTypeDirection !== undefined
      ) {
```